### PR TITLE
Added basic functionality for getting all possible actions for a give…

### PR DESCRIPTION
**Pull request by noahjax**
…n player. Still need to work on devCard actions. Made small change to board.

### CODE REVIEW

File: game.py [hunk 1]
```diff
@@ -112,10 +172,12 @@ def canBuySettlement(self, resources):
             and resources['Wool'] >= 1 and resources['Wheat'] >= 1
 
     def canBuyDevCard(self, resources):
-        if not self.devCards: return False
+        if not self.devCards:
```
Comment (sgoddijn) [inline game.py:175]: Why are you putting this check here? should having a dev card have an effect on whether you can buy one?

File: game.py [hunk 2]
```diff
@@ -198,146 +263,88 @@ def testPiecesPurchasable(self):
...
         #Loop over all edges, check if edge is empty and one of the neighbors is a valid city/road
             #Add valid edges to the list
-        
         return possible_locations
     
-    #Get valid bu
+    '''Missing test for settlement being at end of road'''
+    #Helper to test if node is valid for a settlment. 
+    def isValidSettlement(node):
+        if node.isOccupied or node.neighbors:
```
Comment (sgoddijn) [inline game.py]: does the check for node.neighbors work like this or do we need to check each one to see if it has something in it? the node will always have neighbors I think we need to check if they contain anything

File: game.py [hunk 3]
```diff
@@ -198,146 +263,88 @@ def testPiecesPurchasable(self):
...
         possible_locations = []
 
-        #Loop over all nodes, check if node is settlement
-
-
-
+        #Loop over all nodes, check if there is already a city there with the right owner
+        for row in self.board.nodes.values():
+            for node in row:
+                if node.occupyingPiece is City and node.occupyingPiece.player == player:
```
Comment (sgoddijn) [inline game.py]: I think you meant node.occupyingPiece is Settlement

File: game.py (hunk 1 above)
Comment (noahjax) [reply game.py:175]: Good question where tf did that come from

Comment (noahjax) [reply game.py:175]: Oh wait...this is the game class and devCards are stored in a queue so this check makes sure the queue isn't empty and there are still devCards left to buy

File: game.py (hunk 2 above)
Comment (noahjax) [reply game.py]: yeah you're right

File: game.py (hunk 1 above)
Comment (sgoddijn) [reply game.py:175]: ah got ya, makes sense

**Pull request merged**